File: DSA/Aarohi-Sharma/Day24_q1.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n=nums.size();
        vector<int> arr(n);
        for(int i=0; i<nums.size(); i++){
            arr[i]=pow(nums[i], 2);
        }
        sort(arr.begin(), arr.end());
        return arr;
    }
};

File: DSA/Yashvardhan-Guliyani/Day24_q1.java
class Solution {
    public int[] sortedSquares(int[] arr) {
        int n = arr.length;
        int ans[] = new int[n];
        int idx = n-1;
        int i=0,j=n-1;

        while(i<=j){
            if(Math.abs(arr[i])>Math.abs(arr[j])){
                ans[idx--] = arr[i]*arr[i];
                i++;
            }else{
                ans[idx--] = arr[j]*arr[j];
                j--;
            }
        }
        return ans;
    }
}
